<a href="https://colab.research.google.com/github/Mikhail-068/Scoring/blob/master/%D0%9F%D0%B0%D1%82%D1%8E%D0%BA%D0%BE%D0%B2%D0%B0%20%D0%95.%D0%90./%D0%9F%D0%BB%D0%BE%D1%85%D0%B8%D0%B5_%D0%BA%D0%BB%D0%B8%D0%B5%D0%BD%D1%82%D1%8B.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

In [ ]:
!pip install openpyxl

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


Прочитаем из файла плохие клиенты и строку 2 определим, как строку заголовков.

In [ ]:
data = pd.read_excel('/content/2579_КИдлятеста_плохие_всего4588_безПИ.xlsx', header=2)
data

region_name            city_name  population date of birth  \
0          Санкт-Петербург  г.  Санкт-Петербург     5383890    1991-12-05   
1      Ульяновская область      г. Димитровград      115253    1980-09-21   
2      Ульяновская область      г. Димитровград      115253    1980-09-21   
3      Ульяновская область      г. Димитровград      115253    1980-09-21   
4      Ульяновская область      г. Димитровград      115253    1980-09-21   
...                    ...                  ...         ...           ...   
67558  Ставропольский край      г. Невинномысск      116884    1946-09-15   
67559  Ставропольский край      г. Невинномысск      116884    1946-09-15   
67560  Ставропольский край      г. Невинномысск      116884    1946-09-15   
67561  Ставропольский край      г. Невинномысск      116884    1946-09-15   
67562  Ставропольский край      г. Невинномысск      116884    1946-09-15   

        gender          education                           position  \
0      Мужской  СреднеТехническое  работник промышленных предприятий   
1      Мужской            Среднее  работник промышленных предприятий   
2      Мужской            Среднее  работник промышленных предприятий   
3      Мужской            Среднее  работник промышленных предприятий   
4      Мужской            Среднее  работник промышленных предприятий   
...        ...                ...                                ...   
67558  Женский  СреднеТехническое                          пенсионер   
67559  Женский  СреднеТехническое                          пенсионер   
67560  Женский  СреднеТехническое                          пенсионер   
67561  Женский  СреднеТехническое                          пенсионер   
67562  Женский  СреднеТехническое                          пенсионер   

         dt_nbki                               id_nbki     issued  ...  \
0     2021-01-16  42c79e91-580e-11eb-b807-3cfdfed12971 2021-01-14  ...   
1     2020-11-21  3ae2680a-2bfa-11eb-80c7-f4034344a6bd 2020-04-17  ...   
2     2020-11-21  3ae2680a-2bfa-11eb-80c7-f4034344a6bd 2020-08-21  ...   
3     2020-11-21  3ae2680a-2bfa-11eb-80c7-f4034344a6bd 2018-12-10  ...   
4     2020-11-21  3ae2680a-2bfa-11eb-80c7-f4034344a6bd 2020-11-04  ...   
...          ...                                   ...        ...  ...   
67558 2020-10-30  392179ad-1ab6-11eb-80c7-f4034344a6bd 2014-09-02  ...   
67559 2020-10-30  392179ad-1ab6-11eb-80c7-f4034344a6bd 2012-10-07  ...   
67560 2020-10-30  392179ad-1ab6-11eb-80c7-f4034344a6bd 2012-08-01  ...   
67561 2020-10-30  392179ad-1ab6-11eb-80c7-f4034344a6bd 2011-01-24  ...   
67562 2020-10-30  392179ad-1ab6-11eb-80c7-f4034344a6bd 2011-04-08  ...   

       volume_commitment amount_surety surety_period source_information  \
0                    NaN           0.0           NaT                МФО   
1                    NaN           0.0           NaT               Банк   
2                    NaN           0.0           NaT               Банк   
3                    NaN           0.0           NaT               Банк   
4                    NaN           0.0           NaT                МФО   
...                  ...           ...           ...                ...   
67558                NaN           0.0           NaT               Банк   
67559                NaN           0.0           NaT               Банк   
67560                NaN           0.0           NaT               Банк   
67561                NaN           0.0           NaT               Банк   
67562                NaN           0.0           NaT               Банк   

      maximum_amount_overdue_indebtedness date_credit_state  \
0                                     0.0        2021-01-15   
1                                     0.0        2020-11-18   
2                                     0.0        2020-11-18   
3                                     0.0        2018-12-10   
4                                     0.0        2020-11-16   
...                                   ...               .

В таблице 67563 строки и 41 столбец. Определим уникальные значения по всем столбцам

In [ ]:
for i in data.columns:
  print(i, data[i].nunique())

region_name 69
city_name 674
population 672
date of birth 3960
gender 2
education 7
position 20
dt_nbki 927
id_nbki 4588
issued 5594
amount 12183
currency 2
type_credit 16
status_credit 12
period 7549
period_interest 6867
psk 11051
balance 27198
current_indebtedness 9723
expired 4122
days_overdue 12301
next_payment 5357
last_payment 4938
updated 3784
term_indebtedness_body_loan 6881
term_indebtedness_interest 2717
term_indebtedness_other 942
overdue_indebtedness_body_loan 2641
overdue_indebtedness_interest 1579
overdue_indebtedness_other 784
actual_payments_24_Months 8927
volume_commitment 2
amount_surety 128
surety_period 143
source_information 8
maximum_amount_overdue_indebtedness 2821
date_credit_state 4811
beginning_days_delinquency_counting 4526
type_debt 2
amount_principal_debt_repaid 4402
availability_support 2


Вывод 1. Уникальных значений по датам рождений 3960 - значит столько уникальных клиентов

In [ ]:
data['date of birth'].nunique()

3960

Исследуем столбцы с числовыми данными

Вопрос. Почему выводится в другом формате

In [ ]:
data.describe()

population        amount           psk       balance  \
count  6.756300e+04  6.718900e+04  6.718900e+04  6.718900e+04   
mean   3.249353e+05  3.591734e+04  3.850126e+02  4.177295e+04   
std    7.232128e+05  1.381645e+05  9.069856e+03  9.989535e+05   
min    4.271000e+03  0.000000e+00 -3.251000e+03  0.000000e+00   
25%    3.719700e+04  5.000000e+03  2.389300e+01  1.300000e+03   
50%    1.053840e+05  1.000000e+04  3.560000e+02  8.050000e+03   
75%    3.153110e+05  2.004000e+04  3.650000e+02  2.068900e+04   
max    5.383890e+06  8.000000e+06  1.939059e+06  2.483119e+08   

       current_indebtedness       expired  next_payment  \
count          6.718900e+04  6.718900e+04  6.718900e+04   
mean           1.220512e+04  2.929127e+03  2.218705e+03   
std            8.805972e+04  3.378885e+04  3.250224e+04   
min            0.000000e+00  0.000000e+00  0.000000e+00   
25%            0.000000e+00  0.000000e+00  0.000000e+00   
50%            0.000000e+00  0.000000e+00  0.000000e+00   
75%            0.000000e+00  0.000000e+00  0.000000e+00   
max            7.300431e+06  3.305708e+06  7.492321e+06   

       term_indebtedness_body_loan  term_indebtedness_interest  \
count                 6.718900e+04                67189.000000   
mean                  1.021094e+04                  270.367170   
std                   7.921287e+04                 3528.047559   
min                   0.000000e+00                    0.000000   
25%                   0.000000e+00                    0.000000   
50%                   0.000000e+00                    0.000000   
75%                   0.000000e+00                    0.000000   
max                   6.964782e+06               679873.000000   

       term_indebtedness_other  overdue_indebtedness_body_loan  \
count             6.718900e+04                    6.718900e+04   
mean              1.020398e+02                    1.796609e+03   
std               5.176875e+03                    2.748417e+04   
min               0.000000e+00                    0.000000e+00   
25%               0.000000e+00                    0.000000e+00   
50%               0.000000e+00                    0.000000e+00   
75%               0.000000e+00                    0.000000e+00   
max               1.233490e+06                    2.541982e+06   

       overdue_indebtedness_interest  overdue_indebtedness_other  \
count                   67189.000000                6.718900e+04   
mean                      281.694591                1.282590e+02   
std                      5093.516632                1.204439e+04   
min                         0.000000                0.000000e+00   
25%                         0.000000                0.000000e+00   
50%                         0.000000                0.000000e+00   
75%                         0.000000                0.000000e+00   
max                    421542.000000                2.631099e+06   

       actual_payments_24_Months  amount_surety  \
count               6.718900e+04   6.718900e+04   
mean                7.637826e+03   1.249232e+03   
std                 5.897835e+04   5.359176e+04   
min                 0.000000e+00   0.000000e+00   
25%                 0.000000e+00   0.000000e+00   
50%                 0.000000e+00   0.000000e+00   
75%                 0.000000e+00   0.000000e+00   
max                 8.998162e+06   9.056736e+06   

       maximum_amount_overdue_indebtedness  amount_principal_debt_repaid  \
count                         6.718900e+04                  6.718900e+04   
mean                          1.464320e+03                  7.033548e+03   
std                           2.312710e+04                  1.651662e+05   
min                           0.000000e+00                  0.000000e+00   
25%                           0.000000e+00                  0.000000e+00   
50%                           0.000000e+00                  0.000000e+00   
75%                           0.000000e+00                  0.000000e+00   
max                 

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 67563 entries, 0 to 67562
Data columns (total 41 columns):
 #   Column                               Non-Null Count  Dtype         
---  ------                               --------------  -----         
 0   region_name                          67563 non-null  object        
 1   city_name                            67563 non-null  object        
 2   population                           67563 non-null  int64         
 3   date of birth                        67563 non-null  datetime64[ns]
 4   gender                               67563 non-null  object        
 5   education                            67244 non-null  object        
 6   position                             67245 non-null  object        
 7   dt_nbki                              67563 non-null  datetime64[ns]
 8   id_nbki                              67563 non-null  object        
 9   issued                               67189 non-null  datetime64[ns]
 10  amount    

Из информации видим, что не во всех ячейках ясть данные, т.к. количество ненулевых значений в яче1ках разное в разных столбцах